File: Python/Problems/Problems/problem2856.py
from .problem import Problem


class Problem2856(Problem):
    def test(self):
        nums = [1, 2, 3, 4]

        expected = 0

        result = self.minLengthAfterRemovals(nums)

        print(result)

        return result == expected

    def minLengthAfterRemovals(self, nums):
        n = len(nums)
        if n < 2:
            return n
        left = 0
        right = n - 1
        med = nums[n // 2]
        while nums[left] < med:
            left += 1
        while nums[right] > med:
            right -= 1
        lesser = left
        bigger = n - 1 - right
        equal = n - lesser - bigger
        if lesser + bigger > equal:
            return n % 2

        return equal - lesser - bigger

File: C++/Problems/Problems/problem_2856.cpp
#include "problem_2856.h"

#include <iostream>
#include <ostream>

bool problem_2856::test() {
    std::vector nums  { 1, 2, 3, 4 };

    auto expected = 0;

    auto result = minLengthAfterRemovals(nums);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_2856::minLengthAfterRemovals(std::vector<int> &nums) {
    int n = nums.size();
    if (n < 2)
        return n;
    auto left = 0;
    auto right = n - 1;
    auto med = nums[n / 2];
    while (nums[left] < med)
        left++;
    while (nums[right] > med)
        right--;
    auto lesser = left;
    auto bigger = n - 1 - right;
    auto equal = n - lesser - bigger;
    if (lesser + bigger > equal)
        return n % 2;


    return equal - lesser - bigger;
}
